### PR TITLE
New filter of $args passed to get_posts() in credittracker_get_images()

### LINKED ISSUE #34: Credit Tracker shortcode, display images by tag?
I am using Credit Tracker – love the plugin! – to track the CC licensing for images that we are using on our pages.

Here is an example – https://tellibrary.org/lessons/understanding-mass-communication/
at the bottom under License and Citations.

I am also using a plugin called Enhanced Media Library which allows me to use the word press taxonomy (Categories and Tags) in the Media Library – so every image has a Categories and Tags.

Would it be possible to pull the citations that I need through the shortcode by Tag instead of having to list each library item number?

----

Source: https://wordpress.org/support/topic/credit-tracker-shortcode-display-images-by-tag/

### PR DESCRIPTION
**Pull request by dmchale**
Allow users to modify/remove/add args to get_posts(), allowing more flexibility for custom solutions.

Inspired to help solve #34 , but without requiring a highly specific solution. A filter allows any user to modify the behavior of the query as they see fit.

### CODE REVIEW

File: credit-tracker-functions.php [hunk 1]
```diff
@@ -72,7 +72,7 @@ function credittracker_get_images($attr)
     if (isset($attr['only_current_post']) && true === $attr['only_current_post'] ) {
       $attachments = get_attached_media( 'image', $post->ID );
     } else {
-      $attachments = get_posts($args);
+      $attachments = get_posts( apply_filters( 'credittracker_get_images_args', $args ) );
```
Comment (r-brown) [inline credit-tracker-functions.php:75]: @dmchale 
What will happen here if "regular" user doesn't have `credittracker_get_images_args` defined?

**Comment by dmchale** [comment]
`credittracker_get_images_args` is the tag for the filter. If no code references the filter's tag, the variable (in this case, `$args`) will not be modified and will return itself. So in this case, most of your users will simply pass `$args` to `get_posts()`. Filters create a point in the code where a developer MAY modify the value of something, but it is not required.

https://developer.wordpress.org/reference/functions/apply_filters/ 

Code sample below shows how the filter could be used, which I added to the Issue this PR addresses. A tax query is probably a more complex example, as is only modifying `$args` based on being on a certain page, but I wanted to demonstrate how it may be used in the case of the user who created the issue.

    function my_credittracker_filter( $args ) {

        if ( 12345 == get_queried_object_id() ) {   // Replace 12345 with the postID of the page you wish to run this on, or remove this line and the closing } below to remove
	        $args['tax_query'] = array(
		    array(
			    'taxonomy' => 'media_category',
			    'field'    => 'slug',
			    'terms'    => array( 'my test category' ),   // enter the name(s) of your categories here in the array
			    'operator' => 'IN',
		    )
	        );
        }

        return $args;
    }
    add_filter( 'credittracker_get_images_args', 'my_credittracker_filter', 10, 1 );


A much more simplified example would probably be something like this, where someone just wanted to show 10 posts all the time.

    function my_credittracker_filter( $args ) {

        $args['numberposts'] = 10;

       return $args;
    }
    add_filter( 'credittracker_get_images_args', 'my_credittracker_filter', 10, 1 );

**Comment by r-brown** [comment]
That's great @dmchale - thank you!

**Comment by dmchale** [comment]
You're welcome! I hope some users find it useful

**Pull request merged**